optimize/functions/functions.go

```diff
@@ -482,6 +482,46 @@ func (Box3D) Minima() []Minimum {
 	}
 }
 
+// BraninHoo implements the Branin-Hoo function. BraninHoo is a 2-dimensional
+// test function for global optimization routines. It has three global minima
+// with the same function value. The function is
+//  f(x) = (x_2 - (5.1/(4π^2))*x_1^2 + (5/π)*x_1 - 6)^2 + 10*(1-1/(8π))cos(x_1) + 10
+// It has a minimum value of 0.397887 at x^* = {(-π, 12.275), (π, 2.275), (9.424, 2.475)}
```

Comment:
Please use 9.424778 for the first component of the last minimum, it gives a more accurate function value.